rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -51,8 +51,19 @@ paths:
                   format: binary
                   minLength: 1
                   maxLength: 14316557655
+                correlationId:
+                  description: >
+                    A unique client provided identifier for the product to be ingested which will allow the client to
+                    retrieve the product and/or any other information associated with it in the future. Only one product
+                    can exist with a given correlation identifier. This means that a request to ingest a product with an
+                    existing correlation id will result in a failure with no changes to the existing product or
+                    associated information.
```

Comment:
Here's my attempt at word-smithing:
"The service provider associates the given correlation identify with the product. After the product is ingested, the client is able to retrieve the product and its associated information by searching for the correlation identifier in the system.  The service provider makes a best-effort attempt to ensure no other product is associated with the correlation ID. If the service provider detects the provided correlation ID is already in use, the product will not be ingested. "

Reply:
Careful, we might not be ingesting a product as we might be ingesting metada only. I would reference the id to a piece of intel or something like that. A piece of intel is comprises of maybe a product and maybe one or more metadata document.

❗️ I like the rest except for `makes a best-effort`. It cannot be a best effort. The correlation id is unique or it isn't and in this case it has to be considered unique meaning that CDR must ensure that one correlation id corresponds to only one piece of intel.